### PR TITLE
Fix the `alias` generated when using `bzlmod` and the `maven` resolver

### PR DESCRIPTION
**Pull request by shs96c**
This always assumed that the repo with the default name was visible but what we actually need is the canonical repo name.

### CODE REVIEW

**Review by Wyverald — APPROVED** [review]

File: private/rules/maven_bom.bzl [hunk 1]
```diff
@@ -179,13 +170,9 @@ def maven_bom(
...
-    # Bazel versions, or the older (deprecated) API in Bazel 5. Now, normally
-    # we'd use Skylib's `version` but that needs `native.bazel_version` to be
-    # present, and that's not available to macros. Instead, see whether what
-    # we need is present. Pulling in `bazel_features` for this check seems like
-    # overkill, especially since we'd need to land a patch in it before we
-    # could check this feature. Doing this the Not Invented Here way for now.
-    if "same_package_label" in dir(_NON_EXISTENT_LABEL):
+    # Bazel versions, or the older (deprecated) API in Bazel 5. We use a check
+    # that is a proxy for "is bazel 6 or later"
+    if bazel_features.toolchains.has_optional_toolchains:
```
[the remark this answers is not in the document]
Comment (shs96c) [reply private/rules/maven_bom.bzl]: Oh! Excellent. I'll do that.

[the remark this answers is not in the document]
Comment (shs96c) [reply private/rules/maven_bom.bzl]: Done!

**Review by jin — APPROVED** [review]

**Comment by SanjayVas** [comment]
Does this fix #1205?

**Comment by shs96c** [comment]
I believe to does.

**Comment by guw** [comment]
Does not look like it fixes #1205. Still seeing an issue with 6.6.

**Pull request merged**